Manifold/Term.swift

```diff
@@ -13,6 +13,28 @@ public enum Term: Equatable, IntegerLiteralConvertible, StringLiteralConvertible
 	}
 
 
+	// MARK: Hashable
+
+	public var hashValue: Int {
+		return cata {
+			switch $0 {
+			case let .Type(n):
+				return (Int.max - 59) ^ n
+			case let .Variable(n):
+				return (Int.max - 83) ^ n.hashValue
+			case let .Application(a, b):
+				return (Int.max - 95) ^ a ^ b
+			case let .Lambda(i, t, b):
+				return (Int.max - 179) ^ i ^ t ^ b
+			case let .Embedded(_, _, type):
+				return (Int.max - 189) ^ type
+			case .Implicit:
+				return (Int.max - 257)
+			}
```

Comment:
These magic numbers subtracted from `Int.max` comprise the six largest 64-bit primes.